pkg/minishift/registration/utils.go

```diff
@@ -100,3 +108,111 @@ func UnregisterHostVM(host *host.Host, param *RegistrationParameters) (bool, err
 		registrator.Unregister,
 		param)
 }
+
+func RSHMRegisterAndLoginToRedhatRegistry(sshCommander provision.SSHCommander, param *RegistrationParameters, isRHEL bool) error {
+	var err error
+	keyringDocsLink := "https://docs.okd.io/latest/minishift/troubleshooting/troubleshooting-misc.html#Remove-password-from-keychain"
+	for i := 1; i < 4; i++ {
+		// request username (disallow empty value)
+		if param.Username == "" {
+			// Check if Terminal tty supported or not
+			if !param.IsTtySupported {
```

Comment:
copy-pasted... this is duplication. Can this be extracted in a function that returns `username, password` when succesful, else error?